src/tests/Microsoft.PowerFx.Core.Tests/AttributeParserTests.cs

```diff
@@ -0,0 +1,165 @@
+                [Partial Unknown]
+                Foo = false;
+                [Partial Unknown]
+                Foo = true;
+            ", _parseOptions,
+            out var result);
+
+            Assert.True(result.HasErrors);
+
+            // Combined expression not generated
+            Assert.Equal(2, result.NamedFormulas.Count());
+        }
+
+        [Fact]
+        public void TestMultipleNamedFormulaAttributes()
+        {
+            UserDefinitions.ProcessUserDefinitions(
+            @"
+                [SomeName Ident]
+                Foo = 123;
+
+                [SomeName1 Ident1]
+                Foo1 = 123;
+
+                [SomeName2 Ident2]
+                Foo2 = 123;
+            ", _parseOptions,
+            out var result);
+
+            Assert.False(result.HasErrors);
+
+            var formulas = result.NamedFormulas.ToList();
+
+            Assert.Equal(3, result.NamedFormulas.Count());
+
+            Assert.Equal("Foo", formulas[0].Ident.Name.Value);
+            Assert.Equal("SomeName", formulas[0].Attribute.AttributeName.Name.Value);
+            Assert.Equal("Ident", formulas[0].Attribute.AttributeOperationToken.As<IdentToken>().Name.Value);
+
+            Assert.Equal("Foo1", formulas[1].Ident.Name.Value);
+            Assert.Equal("SomeName1", formulas[1].Attribute.AttributeName.Name.Value);
+            Assert.Equal("Ident1", formulas[1].Attribute.AttributeOperationToken.As<IdentToken>().Name.Value);
+
+            Assert.Equal("Foo2", formulas[2].Ident.Name.Value);
+            Assert.Equal("SomeName2", formulas[2].Attribute.AttributeName.Name.Value);
+            Assert.Equal("Ident2", formulas[2].Attribute.AttributeOperationToken.As<IdentToken>().Name.Value);
+        }
+
+        [Fact]
+        public void TestErrorNamedFormulaAttributes()
+        {
+            UserDefinitions.ProcessUserDefinitions(
+            @"
+                [SomeNa.me Iden()t]
+                Foo = 123;
+
+                [SomeName1 Ident1]
+                Foo1 = 123;
+
+                [SomeName2 Ident2]
+                Foo2 = 123;
+            ", _parseOptions,
+            out var result);
+
+            Assert.True(result.HasErrors);
+
+            Assert.Equal(2, result.NamedFormulas.Count());
+
+            // Error in the first definition's attribute, it gets skipped and we restart on the next one
+            var formulas = result.NamedFormulas.ToList();
+
+            Assert.Equal("Foo1", formulas[0].Ident.Name.Value);
+            Assert.Equal("SomeName1", formulas[0].Attribute.AttributeName.Name.Value);
+            Assert.Equal("Ident1", formulas[0].Attribute.AttributeOperationToken.As<IdentToken>().Name.Value);
+
+            Assert.Equal("Foo2", formulas[1].Ident.Name.Value);
+            Assert.Equal("SomeName2", formulas[1].Attribute.AttributeName.Name.Value);
+            Assert.Equal("Ident2", formulas[1].Attribute.AttributeOperationToken.As<IdentToken>().Name.Value);
+        }
+    }
```

Comment:
>    [](http://example.com/codeflow?start=0&length=3)

It would be good to have a "full" test, where we actually call the combined NF

Reply:
#2186